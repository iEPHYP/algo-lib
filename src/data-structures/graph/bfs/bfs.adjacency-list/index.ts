export * from './bfs.adjacency-list';
