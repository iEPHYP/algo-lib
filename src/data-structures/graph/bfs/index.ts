export * from './bfs.adjacency-list';
export * from './bfs.adjacency-matrix';
