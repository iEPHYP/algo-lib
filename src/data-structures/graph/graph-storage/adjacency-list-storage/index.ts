export * from './adjacency-list.types';
export * from './adjacency-list.utils';
