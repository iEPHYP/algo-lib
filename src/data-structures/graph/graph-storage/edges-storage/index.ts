export * from './edges-storage.types';
export * from './edges-storage.utils';
