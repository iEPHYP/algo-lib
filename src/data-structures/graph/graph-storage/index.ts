export * from './adjacency-list-storage';
export * from './adjacency-matrix-storage';
export * from './edges-storage';
