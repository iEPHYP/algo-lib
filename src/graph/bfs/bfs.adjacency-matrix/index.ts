export * from './bfs.adjacency-matrix';
