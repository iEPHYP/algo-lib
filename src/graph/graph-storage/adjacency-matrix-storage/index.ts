export * from './adjacency-matrix-storage.types';
