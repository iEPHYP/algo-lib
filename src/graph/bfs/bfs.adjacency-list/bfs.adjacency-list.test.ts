import { convertEdgesToAdjacencyListGraph } from '../../graph-storage';
import { indirectConnectedGraphRawEdges } from '../../rawEdges.mock';
import { adjacencyListGraphBfs } from './bfs.adjacency-list';

test('adjacencyListGraphDfs should print visited vertices in the right order', () => {
  const graph = convertEdgesToAdjacencyListGraph(indirectConnectedGraphRawEdges);

  const visitedVertexIndices: number[] = [];

  adjacencyListGraphBfs(graph, (vertex) => {
    visitedVertexIndices.push(vertex.index);
  });

  expect(visitedVertexIndices).toMatchSnapshot();
});
