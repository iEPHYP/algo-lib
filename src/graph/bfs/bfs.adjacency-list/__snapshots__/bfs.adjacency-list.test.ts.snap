// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adjacencyListGraphDfs should print visited vertices in the right order 1`] = `
[
  0,
  1,
  3,
  5,
  6,
  4,
  2,
  7,
]
`;
